pulsar-client/src/main/java/org/apache/pulsar/client/impl/PatternMultiTopicsConsumerImpl.java

```diff
@@ -173,6 +158,11 @@ CompletableFuture<Void> recheckTopicsChange() {
                     return updateSubscriptions(topicsPattern, this::setTopicsHash, getTopicsResult,
                             topicsChangeListener, oldTopics, subscription);
                 }
+            }).thenAccept(__ -> {
+                if (recheckPatternTimeout != null) {
```

Comment:
After the change here, even if enabled `broker -> enableBrokerSideSubscriptionPatternEvaluation`, this task will also be scheduled triggered, which is unnecessary.

Reply:
if enableBrokerSideSubscriptionPatternEvaluation, the recheckPatternTimeout  will == null.

I placed the initialization of recheckPatternTimeout in the exception of watcherFuture.

Reply:
Sure, we'd better add a comment for this variable to mark that it has two meanings